nomic/dataset.py

```diff
@@ -156,10 +161,74 @@ def _get_project_by_id(self, project_id: str):
         )
 
         if response.status_code != 200:
-            raise Exception(f"Could not access project with id {project_id}: {response.text}")
+            raise Exception(f"Could not access dataset with id {project_id}: {response.text}")
+
+        return response.json()
+
+    def _get_organization_by_slug(self, slug: str):
+        '''
+
+        Args:
+            slug: The organization slug
+
+        Returns:
+            Returns the requested project.
```

Comment:
Would be better to merge upstream and just hit f"/v1/organization/{slug}" for full organization info directly, avoiding a roundtrip.
```suggestion
            An organization id
```